include/nuttx/spinlock.h

```diff
@@ -626,6 +626,81 @@ irqstate_t spin_lock_irqsave(FAR volatile spinlock_t *lock)
 #  define spin_lock_irqsave(l) ((void)(l), up_irq_save())
 #endif
 
+/****************************************************************************
+ * Name: spin_trylock_irqsave_wo_note
+ *
+ * Description:
+ *   Try once to lock the spinlock.  Do not wait if the spinlock is already
+ *   locked.
+ *
+ *   This implementation is the same as the above spin_trylock() except that
+ *   it does not perform instrumentation logic.
+ *
+ * Input Parameters:
+ *   lock  - A reference to the spinlock object to lock.
+ *   flags - flag of interrupts status
+ *
+ * Returned Value:
+ *   SP_LOCKED   - Failure, the spinlock was already locked
+ *   SP_UNLOCKED - Success, the spinlock was successfully locked
+ *
+ * Assumptions:
+ *   Not running at the interrupt level.
+ *
+ ****************************************************************************/
+
+#ifdef CONFIG_SPINLOCK
+#  define spin_trylock_irqsave_wo_note(l, f) \
+({ \
```

Comment:
but not all compiler support the expression statement.

Reply:
Only MSVC does not support this syntax, which is  simulation environment and does not require spinlock support.

Reply:
if sim on windows support SMP, this code will break windows porting.

Reply:
There is already a similar implement in sim, so are xiaomi still using MSVC in windows?
https://github.com/apache/nuttx/blob/master/arch/sim/src/sim/sim_internal.h#L127-L135

Reply:
There is already a similar implement in sim, so are xiaomi still using MSVC in windows?
https://github.com/apache/nuttx/blob/master/arch/sim/src/sim/sim_internal.h#L127-L135

Reply:
I don' t see there is any issue to block use inline function.

Reply:
If similar code in xiaomi internal repository can be compiled normally, it means that there is no issue with the `expression statement` on MSVC. could we merge this PR ?